Stratis.Bitcoin.Features.BlockStore/LoopSteps/CheckNextChainedBlockExistStep.cs

```diff
@@ -31,10 +31,10 @@ internal CheckNextChainedBlockExistStep(BlockStoreLoop blockStoreLoop, ILoggerFa
         /// <inheritdoc/>
         internal override async Task<StepResult> ExecuteAsync(ChainedBlock nextChainedBlock, CancellationToken cancellationToken, bool disposeMode)
         {
-            this.logger.LogTrace("({0}:'{1}/{2}',{3}:{4})", nameof(nextChainedBlock), nextChainedBlock?.HashBlock, nextChainedBlock?.Height, nameof(disposeMode), disposeMode);
-
             if (await this.BlockStoreLoop.BlockRepository.ExistAsync(nextChainedBlock.HashBlock))
             {
+                this.logger.LogTrace("({0}:'{1}/{2}',{3}:{4})", nameof(nextChainedBlock), nextChainedBlock?.HashBlock, nextChainedBlock?.Height, nameof(disposeMode), disposeMode);
+
```

Comment:
Why this change? First, the log should be on the method entry, not after the first if. Second, there should be log on each method exit.

Reply:
I really only want the log to output something if the logic contained in it became applicable. I'll remove the method entry format from the log. I ran the node with the log and it output thousands of useless method entry and exit logs which wasn't useful at all.

Reply:
Then consider DEBUG level, but don't remove or change TRACE level. I understand that for lot of tasks, you are not interested in seeing entry/exit logs, but for other tasks, other devs might, so if you want to have something cherrypicked, use DEBUG level and then modify your NLog.config in a way that you only see DEBUG level from that module. This will perfectly fit what you need while you won't harm the general trace logging.

Reply:
Ok yes, that makes sense! Ok Ive added them back and changed the other ones to Debug.